### PR TITLE
Source FB Marketing: add incremental support

### LINKED ISSUE #1478: Source Facebook Marketing: Add incremental update
## Tell us about the problem you're trying to solve
Original singer tap (https://github.com/singer-io/tap-facebook) supports incremental update (bookmarks)

## Describe the solution you’d like
Support save state and restore state for all streams

## Describe the alternative you’ve considered or used
Using singer tap

## Additional context

The estimation:
- [ ]  Implement passing filters to requests, `get_state` and `set_state` methods  - 5h
- [ ]  Check if testing is possible with automated tests - 15m
- [ ]  Perform manual tests with each stream - 1h
Dev Total: ~6h

### LINKED ISSUE #1682: Broken links in FB docs
I deleted some links out of the facebook docs in this PR: https://github.com/airbytehq/airbyte/pull/1681. The links were broken and causing the build to fail. Looks like when the FB source got updated ([commit](https://github.com/airbytehq/airbyte/commit/fb7262dfa8a76233a4a924b650b9a00594e2423e)) some screenshots didn't get included in source. We should add those back so that the docs make sense.

### PR DESCRIPTION
**Pull request by eugene-kulak**
closes https://github.com/airbytehq/airbyte/issues/1682

## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
*Describe the solution*

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

## Recommended reading order
1. `test.java`
1. `component.ts`
1. the rest

## Tested scenarios
- [x] sync each stream with `incremental`/`full` method
- [x] sync stream with `incremental` method and `state.json` file

### CODE REVIEW

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/client/api.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    def _get_campaigns(self, params):
+        """Separate method to request list of campaigns
+        This is necessary because the functions that call this endpoint return
+        a generator, whose calls need decorated with a backoff.
+        """
+        return self._api.account.get_campaigns(params={**params, **self._state_filter()}, fields=[self.state_pk])
+
+
+#
+# class AdsInsightAPI(IncrementalStreamAPI):
```
Comment (jrhizor) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/client/api.py:229]: Can you add a comment for why this is disabled in the code?

**Review by jrhizor — CHANGES_REQUESTED** [review]
Generally looks good.

The integration test is failing. Once that's fixed please leave a comment and I'll take another look.

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.json [hunk 2]
```diff
@@ -21,11 +21,6 @@
         "type": "string",
         "description": "The value of the access token generated. See the <a href=\"https://docs.airbyte.io/integrations/sources/facebook-marketing\">docs</a> for more information",
         "airbyte_secret": true
-      },
-      "include_deleted": {
```
[the remark this answers is not in the document]
Comment (eugene-kulak) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.json:25]: it doesn't work yet, there is a ticket to add deleted records

**Comment by eugene-kulak** [comment]
@jrhizor please continue
```
Test run finished after 81748 ms
[         2 containers found      ]
[         0 containers skipped    ]
[         2 containers started    ]
[         0 containers aborted    ]
[         2 containers successful ]
[         0 containers failed     ]
[         7 tests found           ]
[         0 tests skipped         ]
[         7 tests started         ]
[         0 tests aborted         ]
[         7 tests successful      ]
[         0 tests failed          ]
```

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/test_api.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        api = AdCreativeAPI(api=api_mock)
+        api.list(fields=[])
+
+
+class TestCampaignAPI:
+    def test__init(self):
+        api_mock = Mock()
+        api_mock.account.get_ad_creatives.return_value = [1, 2, 3]
+
+        api = AdCreativeAPI(api=api_mock)
```
Comment (jrhizor) [inline airbyte-integrations/connectors/source-facebook-marketing/unit_tests/test_api.py:42]: Flake8 is failing to lint this due to:
```
[python] .venv/bin/python -m flake8 . --config /home/runner/work/airbyte/airbyte/tools/python/.flake8 |  
-- | --
  | ./unit_tests/test_api.py:42:9: F841 local variable 'api' is assigned to but never used |  
  | ./unit_tests/test_api.py:45:9: F821 undefined name 'api' |  
  | ./unit_tests/test_api.py:48:9: F821 undefined name 'api'
```

**Comment by eugene-kulak** [comment]
@jrhizor I had to do force push because one commit from `source-file` PR was added in this PR by accident

**Pull request merged**